Avatar Toolkit Overhaul - Do not merge yet

RinaDev-xyz opened the pull request:
I'm have started an overhaul of Avatar Toolkit to enhance stability, maintainability, and organization. Once this PR has been merge it should mean the first Alpha release can happen. Therefor please do not merge unto I have gotten everything done.

### Key Improvements so far:

#### Plugin Registration System:
- Implemented a modern auto-registration system
- Eliminated manual @register_wrapper decorators in favor of automatic class detection
- Removed legacy "bpy" local variable checks

#### New Auto-loader Implementation
- Added auto_load.py for robust module management, more modern approach and less error prone.
- Automatic module discovery and dependency resolution
- Optimized registration/unregistration sequence through topological sorting

#### File Structure Reorganization
- I started to add additional folders, the plugin is getting larger and larger and I don't like having 30 files in one folder. Importers have been moved to their own folder, same with exporters in the core folder.
- I will be doing more code and file re-organization.
- Translation has been moved to core folder as it's more of a core system.

There are more changes, and more changes needed to get all the functions working again seeing I basically changed a core component so late on, but it's looking better then before.

Initial implementation shows promising results, but  testing is still in progress.

RinaDev-xyz committed:
Plugin Registration Changes

- Re-wrote how the plugin registers itself.
- No longer need @register_wrapper classes get auto detected and added.
- The new Auto loader is much better then the old way, no longer need "if "bpy" not in locals():" this was an old way of doing things and wasn't really efficient.

 using auto_load.py provides several advantages:

- It automatically discovers and loads all modules in the addon.
- It handles dependencies between classes correctly through topological sorting.
- It manages registration order automatically.
- It properly handles unregistration in the correct order.

This approach is much less error prone and I not had any issues so far. However it still needs testing fully.

I have also start to re-organise files into folders as well, this is going to be needed so we don't have a long list of files as Avatar Toolkit is getting larger then i originally planned.

RinaDev-xyz committed:
Fixes

This fixes is to get everything working on the new auto load and properties system.
Also some other small fixes.

RinaDev-xyz committed:
Fixes

RinaDev-xyz committed:
Start of the Major Overhaul

I decided to go through each function and UI section one by one, improving and overhauling things. Each function and section is going to be fully tested and not rushed out.

This is the best way to catch things, but also include the code base as much as possible.

Yusarina commented:
I decided to go through each function and UI section one by one, improving and overhauling things. Each function and section is going to be fully tested and not rushed out. 

This is the best way to catch things, but also include the code base as much as possible. This mean most functions have been removed unto I go through them, there will be added back slowly. 

I truly want the overhaul to be the best the toolkit can offer so we can finally get the Alpha out. The plan is as I stated before that this overhaul will end up being the first Alpha.

RinaDev-xyz committed:
Pose mode improvements, armature validation improvements.

Pose mode Improvements:

Batch processing for all mesh operations
Numpy-powered vertex array handling
Optimized modifier stack management
Smart shape key processing
Enhanced progress tracking

The armature validation system improvements:

Essential bones (hips, spine, chest, neck, head)
Proper bone hierarchy validation
Symmetry pair verification (e.g., arm.l/arm.r)

RinaDev-xyz committed:
Setting Panel Added, Debuging Added.

Added the Armature Validation modes now, we have Stritct, Basic and None, it will give a warning to the user in the panel if there have it set to basic or none.
Settings panel added, langauge change has been added back. Did some work on it to slightl improve the system.
Added dubug area, basically everything but autoload will use logging now, you be able to turn it on/off in debug settings.
Did other bits and bobs.

RinaDev-xyz committed:
Optimzation Panel Re-Added

- Major Improvements all round, Join Meshes improved, Combined Materials, Remove Doubles Improvements.

RinaDev-xyz committed:
Start of Tools Panel

Several Improvements and etc. Still need to do the other half of the functions but getting there.

RinaDev-xyz committed:
Tools Panel Finished.

RinaDev-xyz committed:
Start of MMD Tools

- The idea is to have several buttons which kinda mimic what Cats used to do.
- These are very basic, don't work very well, will improve before Alpha 1.

RinaDev-xyz committed:
Re-do still hate it

RinaDev-xyz committed:
Re-do 3rd attempt I hate MMD stuff

RinaDev-xyz committed:
Fixes

RinaDev-xyz committed:
Added Eye tracking and Visemes

RinaDev-xyz committed:
Eye tracking fixes

RinaDev-xyz committed:
Armature Merging

RinaDev-xyz committed:
Fixes

RinaDev-xyz committed:
Attach Meshes

RinaDev-xyz committed:
Finale Fixes
